hw4/sorting/main.go

```
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"time"
+)
+
+type Person struct {
+	firstName, lastName string
+	birthDay time.Time
+}
+
+type People []Person
+
+func (p People) Len() int {
+	return len(p)
+}
+
+func (p People) Less(i, j int) bool  {
+	if p[i].birthDay == p[j].birthDay {
+		return p[i].firstName+p[i].lastName < p[j].firstName+p[j].lastName
+	}
+	return p[i].birthDay.After(p[j].birthDay)
+}
+
+func (p People) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+func (p People) String() (res string) {
+	for i, v := range p {
+		res += fmt.Sprintf("%v %s %s (%v)\n", i+1, v.firstName, v.lastName, v.birthDay.Format("2006-1-2"))
+	}
+	return res
+}
+
+func stringToDateConverter(s string) time.Time {
+	date, err := time.Parse("2006-1-2", s)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Will be better to return an error as a second return value.

---

I'm on the same page with Ivan, return an error as a second argument or as an exceptional case use panic but os.Exit is not obvious for all.

---

Fixed)